MSAL/test/unit/MSALTelemetryTests.m

```diff
@@ -0,0 +1,119 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSALTestCase.h"
+#import "MSALTelemetryAPIEvent.h"
+#import "MSALTelemetry.h"
+#import "MSALTelemetry+Internal.h"
+#import "MSALTelemetryHttpEvent.h"
+#import "MSALTelemetryEventStrings.h"
+#import "MSALTelemetryTestDispatcher.h"
+#import "MSALTelemetryDefaultEvent.h"
+
+@interface MSALTelemetryTests : MSALTestCase
+
+@property (nonatomic) NSArray<NSDictionary<NSString *, NSString *> *> *receivedEvents;
+@property (nonatomic) MSALTelemetryTestDispatcher *dispatcher;
+
+@end
+
+@implementation MSALTelemetryTests
+
+- (void)setUp
+{
+    [super setUp];
+    
+    self.dispatcher = [MSALTelemetryTestDispatcher new];
+    
+    [[MSALTelemetry sharedInstance] addDispatcher:self.dispatcher setTelemetryOnFailure:NO];
+    
+    __weak MSALTelemetryTests *weakSelf = self;
+    [self.dispatcher setDispatcherCallback:^(NSArray<NSDictionary<NSString *, NSString *> *> *event)
+     {
+         weakSelf.receivedEvents = event;
+     }];
+    
+    [MSALTelemetry sharedInstance].piiEnabled = NO;
+}
+
+- (void)tearDown
+{
+    [super tearDown];
+    
+    self.dispatcher = nil;
+    
+    [MSALTelemetry sharedInstance].piiEnabled = NO;
+}
+
+- (void)test_telemetryPiiRules_whenPiiEnabledNo_shouldSetPiiFieldsToEmpty
+{
+    [MSALTelemetry sharedInstance].piiEnabled = NO;
+    NSString *requestId = [[MSALTelemetry sharedInstance] telemetryRequestId];
+    NSString *eventName = @"test event";
+    MSALTelemetryDefaultEvent *event = [[MSALTelemetryDefaultEvent alloc] initWithName:eventName context:nil];
+    [event setProperty:MSAL_TELEMETRY_KEY_DEVICE_IP_ADDRESS value:@"192.168.0.1"];
```

Comment:
Should we delete the IP address all together like in ADAL? In that case those tests will need to be changed.

Reply:
Deleted IP address and hashed PII fields in separate PR.